lib/Driver/ToolChains.h

```diff
@@ -96,12 +96,20 @@ class LLVM_LIBRARY_VISIBILITY GenericUnix : public ToolChain {
   ~GenericUnix() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Android : public GenericUnix {
+protected:
+  std::string getTargetForLinker() const override;
+
+  bool shouldProvideRPathToLinker() const override;
+public:
+  Android(const Driver &D, const llvm::Triple &Triple) : GenericUnix(D, Triple) {}
+  ~Android() = default;
+};
+
 class LLVM_LIBRARY_VISIBILITY Cygwin : public GenericUnix {
 protected:
   std::string getDefaultLinker() const override;
 
-  bool shouldSpecifyTargetTripleToLinker() const override;
```

Comment:
Ah, nice catch, thanks! (I should really get a Cygwin environment so that I can catch these myself...)

Looking into adding tests now.